beacon-chain/rpc/eth/beacon/sync_committee.go

```diff
@@ -2,13 +2,95 @@ package beacon
 
 import (
 	"context"
+	"sort"
+	"strconv"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/altair"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/rpc/statefetcher"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
 	"github.com/prysmaticlabs/prysm/proto/eth/v2"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-func (bs *Server) ListSyncCommittees(ctx context.Context, request *eth.StateSyncCommitteesRequest) (*eth.StateSyncCommitteesResponse, error) {
-	panic("implement me")
+func (bs *Server) ListSyncCommittees(ctx context.Context, req *eth.StateSyncCommitteesRequest) (*eth.StateSyncCommitteesResponse, error) {
+	var st state.BeaconState
+	if req.Epoch != nil {
+		slot, err := helpers.StartSlot(*req.Epoch)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not calculate start slot for epoch %d: %v", *req.Epoch, err)
+		}
+		st, err = bs.StateFetcher.State(ctx, []byte(strconv.FormatUint(uint64(slot), 10)))
+		if err != nil {
+			if stateNotFoundErr, ok := err.(*statefetcher.StateNotFoundError); ok {
+				return nil, status.Errorf(codes.NotFound, "State not found: %v", stateNotFoundErr)
+			} else if parseErr, ok := err.(*statefetcher.StateIdParseError); ok {
+				return nil, status.Errorf(codes.InvalidArgument, "Invalid state ID: %v", parseErr)
+			}
+			return nil, status.Errorf(codes.Internal, "Invalid state ID: %v", err)
```

Comment:
Maybe better to refactor this to something like `handleStateError` then reuse it at L41? I think there's more places like this too

Reply:
+1, I was about to do this in a separate PR but might as well add it to this one.